### PR TITLE
Tile improvements and fixes

### LINKED ISSUE #1299: [component] chart improvements
- [x] Fullscreen view / zoomable view -> open in popup
- [x] show title, legend
- [x] show date in tooltip
- [x] more datasets in one chart
- [x] Barcharts, StackedBarCharts
- [x] chart as background of a tile
- [x] chart without axes
- [x] scroll in charts
- [x] document charts

### PR DESCRIPTION
**Pull request by peuter**
Collection of features and fixes. some of them are general like adding more options to the screenshot generation (and also fix some problems there), augmenting the tile-demo, fix some minor problems that popped up during development and so on.

New features are:

- Charts: bar charts, series selection and navigation through history -> closes #1299 
- State-Notifications (basically like the existing ones, added possibility to manipulate the via the internal system backend)
- New Widget: <cv-status-chart> that combines the known Status-Widget with a background chart that is opened in a popup when clicked on it (see demo)
- Added a generic cached fetch function that allows e.g. using the same chart data for several widgets and make sure that it is only loaded once
- Fixing and improving the known way of getting fake test data into the demo configs (also use this partially for creating screenshots e.g. for charts)

Breaking changes:
There is a wrapper for cv-tile now called cv-widget. This is more in sync with how template widgets are structured and allows generic header and footer inside a widget.

### CODE REVIEW

File: doc/manual/de/config/structure-tile/index.rst [hunk 1]
```diff
@@ -1,13 +1,11 @@
...
-zusammen.
-
-Für häufig benötigte Dinge liefert die Tile-Struktur bereits Kacheln mit vor-definiertem Inhalt mit (im folgenden
-Widgets genannt). So enthält das Switch-Widget z.B. einen Button in der mittleren Zelle und zentrierten Text in der Zeile darunter.
-
-Eine Konfigurationsdatei folgt grundsätzlich folgendem Aufbau:
+Auf der obersten Ebene enthält eine Tile-Datei zunächst einen Meta-Bereich (``<cv-meta>``) in dem nicht sichtbare Einstellungen
+enthalten sind die für diese Konfigurationsdatei benötigt werden (z.B. Verbindungen zu Backends, das Laden von
+zusätzliche Dateien usw.).
+Der sichtbare gliedert sich in einen ``<header>`` Bereich (Kopfzeile) einen ``<main>``-Bereich (der eigentliche Inhalt)
```
Comment (ChristianMayer) [inline doc/manual/de/config/structure-tile/index.rst:7]: Klingt komisch / unvollständig

**Comment by ChristianMayer** [comment]
Pulling this PR is breaking the loading. It is compiled with `--clean` and also the offline files in the browser.

The stack trace isn't looking helpful to me, but probably it's helping you:
grafik
grafik
grafik

**Comment by peuter** [comment]
Did you load a special config file when that error occurs? Or does it occur all the time?

**Comment by ChristianMayer** [comment]
Hm, the posted issue above has gone and didn't come again. Let's think of it as a cache issue.

Trying the demo-tile config is see these issues:

- Clicking on the little triangle the background sticks out of the container, but even worse: I'd expect the "more charts" pop up to disappear when clicking somewhere but it keeps staying open
grafik
Generally the navigating with "more charts" isn't as smooth as expected. Going to "widgets" and there on the "More charts" tile I get lots of strange horizontal lines and see no way of navigating back (except the menu bar on the bottom):
grafik
- Clicking here 
grafik
shows a diagram that seems wrongly scaled: 
grafik
(Please note: I'm using a HiDPI screen with 4K@17", which might be relevant here)
The diagram pop ups of "Zählerstand Gas" and "Zählerstand Wasser" are looking fine

**Comment by ChristianMayer** [comment]
I also get quite a lot of console messages like this:
grafik
I think those aren't needed anymore when pulled

**Comment by peuter** [comment]
Found issues should be fixed now.

Regarding the chart in fullscreen view: Please keep in mind that this is just a zoomed view of the already rendered chart, so its something different that the popup charts you get when you click e.g.. on the gas-chart. Those are seperate HTML-Elements. The fullscreen is just zooming in the shown HTML-Element. Thats why the lines/texts are usually bigger, because its just a zoom. But anyways the zoomed chart should never exceed the boundaries of the surrounding container. That has been fixed.

**Comment by peuter** [comment]
> ... and see no way of navigating back (except the menu bar on the bottom)

Thats currently on purpose that you use the menu for navigating back. I might come up with a better solution, when I augment the demo-config. For my personal use I do not need a better one, because I do not need (and like) having such a navigation depth. Lets face the problem when the demo-tile config is done and has more pages. A breadcrumb menu might be needed then.

**Comment by ChristianMayer** [comment]
Issues are fixed now :+1: 

But there's a new one: clicking on the menu bar little triangle to make "Mehr Diagramme" show does work. But then clicking on the "Mehr Diagramme" most of the times does not do anything and thus does not move to that page.

**Comment by peuter** [comment]
I haven't changed anything there, that problem must have existed before. I could not reproduce the problem, when  I clicked on the text is always worked (chrome and firefox). But when you clicked on the area around the text it did not.
I changed that behaviour now, the whole menu item should react to the click now.

**Comment by ChristianMayer** [comment]
It's still a problem. Probably it was also a problem before - but I couldn't test it as the demo-tile config didn't have such a sub menu before this PR.

Clicking of the sub menu I can see that the call goes till line 218 of `source/class/cv/ui/structure/tile/components/Menu.js`. I don't know whether that is the correct code path as it doesn't look to me like it is supposed to trigger a page change.

**Comment by peuter** [comment]
Thats just the code for closing the menu, the link is pure HTML code a `<a href="#more-charts` do you see the URL change in your browser when clicking the menu item?

**Comment by ChristianMayer** [comment]
@peuter Did my information in Gitter reach you?

**Comment by peuter** [comment]
I got no notification from gitter about new messages. But the messages are there

**Review by ChristianMayer — APPROVED** [review]

**Pull request merged**